src/shared/hooks/useModal.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useEffect, useState } from "react"
+
+interface UseModalProps {
+  openOnMount?: boolean
+}
+
+export const useModal = ({ openOnMount = false }: UseModalProps = {}) => {
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false)
+
+  const handleModalOpen = useCallback(() => {
+    setIsModalOpen(true)
+  }, [])
+
+  const handleModalClose = useCallback(() => {
+    setIsModalOpen(false)
+  }, [])
+
+  useEffect(() => {
+    if (openOnMount) {
+      handleModalOpen()
+    }
+  }, [openOnMount, handleModalOpen])
+
+  return {
+    isModalOpen,
+    handleModalOpen,
+    handleModalClose,
+  }
+}
```

Comment:
```suggestion
export const useModal = ({ openOnMount = false }: UseModalProps = {}) => {
  const [isModalOpen, setIsModalOpen] = useState<boolean>(openOnMount)

  const handleModalOpen = useCallback(() => {
    setIsModalOpen(true)
  }, [])

  const handleModalClose = useCallback(() => {
    setIsModalOpen(false)
  }, [])

  return {
    isModalOpen,
    handleModalOpen,
    handleModalClose,
  }
}
```

모달 on, off 상태에 대한 관리를 훅으로 만드셨네요!! 👍 초기값을 인자로 받을 수 있는 형태라면 useEffect없이 state의 초기값으로 할당하면 어떨까요?!